src/mysky/encrypted_files.test.ts

```diff
@@ -0,0 +1,145 @@
+      }
+    }
+    return { pass: true, message: () => `expected ${received} not to equal ${argument}` };
+  },
+});
+
+describe("deriveEncryptedFileKeyEntropy", () => {
+  it("Should derive the correct encrypted file key entropy", () => {
+    // Hard-code expected value to catch breaking changes.
+    const pathSeed = "a".repeat(64);
+    const expectedEntropy = [
+      145, 247, 132, 82, 184, 94, 1, 97, 214, 174, 84, 50, 40, 0, 247, 144, 106, 110, 227, 25, 193, 138, 249, 233, 32,
+      94, 186, 244, 48, 171, 115, 171,
+    ];
+
+    const result = deriveEncryptedFileKeyEntropy(pathSeed);
+
+    expect(result).toEqualUint8Array(new Uint8Array(expectedEntropy));
+  });
+});
+
+describe("deriveEncryptedFileSeed", () => {
+  it("Should derive the correct encrypted file seed", () => {
+    // Hard-code expected value to catch breaking changes.
+    const pathSeed = "a".repeat(64);
+    const subPath = "path/to/file.json";
+
+    // Derive seed for a file.
+    const fileSeed = deriveEncryptedFileSeed(pathSeed, subPath, false);
+
+    expect(fileSeed).toEqual("ace80613629a4049386b3007c17aa9aa2a7f86a7649326c03d56eb40df23593b");
+
+    // Derive seed for a directory.
+    const directorySeed = deriveEncryptedFileSeed(pathSeed, subPath, true);
+
+    expect(directorySeed).toEqual("fa91607af922c9e57d794b7980e550fb15db99e62960fb0908b0f5af10afaf16");
+
+    expect(fileSeed).not.toEqual(directorySeed);
+  });
+});
+
+describe("deriveEncryptedFileTweak", () => {
+  it("Should derive the correct encrypted file tweak", () => {
+    // Hard-code expected value to catch breaking changes.
+    const seed = "test.hns/foo";
+    const expectedTweak = "352140f347807438f8f74edf3e0750a408f39b9f2ae4147eb9055d396b467fc8";
+
+    const result = deriveEncryptedFileTweak(seed);
+
+    expect(result).toEqual(expectedTweak);
+  });
+});
+
+const encryptedTestFilePath = "test_data/encrypted-json-file";
+const json = { message: "text" };
+const v = ENCRYPTED_JSON_RESPONSE_VERSION;
+const fullData = { _data: json, _v: v };
+const key = new Uint8Array(ENCRYPTION_KEY_LENGTH);
+
+describe("decryptJSONFile", () => {
+  it("Should decrypt the given test data", () => {
+    const data = new Uint8Array(readFileSync(encryptedTestFilePath));
+    expect(data.length).toEqual(4096);
+
+    const result = decryptJSONFile(data, key);
+
+    expect(result).toEqual(fullData);
+  });
+
+  it("Should fail to decrypt bad data", () => {
+    expect(() => decryptJSONFile(new Uint8Array(4096), key)).toThrowError(
+      "Could not decrypt given encrypted JSON file"
+    );
+  });
+});
+
+describe("encryptJSONFile", () => {
+  const result = encryptJSONFile(fullData, key);
+
+  expect(result.length).toEqual(4096);
+});
+
+describe("encodeEncryptedFileMetadata", () => {
+  it("Should fail to encode metadata with an invalid version", () => {
+    const version = 256;
+    const metadata = { version };
+    expect(() => encodeEncryptedFileMetadata(metadata)).toThrowError(
```

Comment:
This test made me realize I haven't seen any version validation. Wouldn't we want to validate we're seeing the expected version when we decrypt the JSON?